File: Intents/app/src/main/java/com/example/intents/MainActivity.java
package com.example.intents;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    EditText txt1, txt2;
    Button b1, b2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        txt1=findViewById(R.id.et1);
        txt2=findViewById(R.id.et2);
        b1=findViewById(R.id.bt1);
        b2=findViewById(R.id.bt2);

        b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String fname= txt1.getText().toString();
                String lname = txt2.getText().toString();
                Intent intent = new Intent(getApplicationContext(),WelcomeActivity.class);
                intent.putExtra("fn", fname);
                intent.putExtra("ln", lname);
                startActivity(intent);
            }
        });
        b2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                txt1.setText("");
                txt2.setText("");
            }
        });
    }
}

File: animation/app/src/main/java/com/example/animation/MainActivity.kt
package com.example.animation

import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        val fadebtn = findViewById<Button>(R.id.fadebtn)
        val zoombtn = findViewById<Button>(R.id.zoombtn)
        val slidebtn = findViewById<Button>(R.id.slidebtn)
        val flashbtn = findViewById<Button>(R.id.flashbtn)
        val spinbtn = findViewById<Button>(R.id.spinbtn)
        val dropbtn = findViewById<Button>(R.id.dropbtn)

        fadebtn.setOnClickListener {
            val animation=AnimationUtils.loadAnimation(this@MainActivity,R.anim.fade)
            fadebtn.startAnimation(animation) }
        zoombtn.setOnClickListener {
            val animation = AnimationUtils.loadAnimation(this@MainActivity,R.anim.zoom)
            zoombtn.startAnimation(animation) }
        slidebtn.setOnClickListener {
            val animation = AnimationUtils.loadAnimation(this@MainActivity,R.anim.slide)
            slidebtn.startAnimation(animation) }
        flashbtn.setOnClickListener {
            val animation = AnimationUtils.loadAnimation(this@MainActivity,R.anim.flash)
            flashbtn.startAnimation(animation) }
        spinbtn.setOnClickListener {
            val animation = AnimationUtils.loadAnimation(this@MainActivity,R.anim.spin)
            spinbtn.startAnimation(animation) }
        dropbtn.setOnClickListener {
            val animation = AnimationUtils.loadAnimation(this@MainActivity,R.anim.drop)
            dropbtn.startAnimation(animation) }
    }
}